src/super_gradients/compliance/check_compliance.py

```diff
@@ -0,0 +1,122 @@
+"""
+A Python script that check's the compliance of Hosts (OS and Hardware) for Deci's infrastructure.
+"""
+
+import os
+import sys
+from pip._internal.operations.freeze import freeze
+from logging import getLogger, DEBUG
+from typing import List, Dict
+from pathlib import Path
+from packaging.version import Version
+
+logger = getLogger('deci-compliance')
+logger.setLevel(DEBUG)
+
+
+def verify_os() -> List[str]:
+    """Verifying operating system name and platform"""
+    name = os.name
+    platform = sys.platform
+    logger.info(f'OS Name: {name}')
+    logger.info(f'OS Platform: {platform}')
+    if 'linux' not in platform.lower():
+        return ['Deci officially supports only Linux kernels. Some features may not work as expected.']
+    return []
+
+
+def get_libs_requirements() -> List[str]:
+    """Read requirement.txt from the root, and split it as a list of libs/version"""
+    with open(Path(__file__).parent.parent.parent.parent / "requirements.txt", "r") as f:
```

Comment:
I am sure that there is a nicer way than .parent.parent.parent.parent
perhaps use pkg_resources

Reply:
I didnt find any satisfying solution using pkg_resources, did you have something in mind ?
What blocked me is that pkg_resources refers to the python package "super-gradients/src/super_gradients" and not to the root "super-gradients" which includes the "requirements.txt" file.

The other approach I can think of is to use the path from where the script is run and then move to "requirements.txt". This is great if we run from the root, but if it is run from somewhere else then the path will change.
For this reason I prefered to refer to the path of "requirements.txt" using the compliance script path as a reference, and then move to the requirements path using ".parent".

I just splited "Path(__file__).parent.parent.parent.parent / "requirements.txt"" into 3 subpart to explicit a bit more what is done.

Reply:
wait... so this will not work if SG was installed using pip?

Reply:
It will work now. I added a step in the ci to copy the requirements.txt into the source code and also added it to the setup.py to have it installed along with the rest of the package